internal/core/src/index/ScalarIndex.h

```diff
@@ -31,6 +31,12 @@ class ScalarIndex : public IndexBase {
     virtual const TargetBitmapPtr
     NotIn(size_t n, const T* values) = 0;
 
+    virtual const TargetBitmapPtr
+    EvalEq(std::string arith_op, T right_operand, T value) = 0;
```

Comment:
I don't think `EvalEq` and `EvalNotEq` should be defined in an index. Maybe we should use raw-data to do these two filterings. As you have already implemented, in fact, many index type can't support them.

Reply:
RIght! I removed this in the updated version. I followed your advice of processing the data directly using the `elem_func` construct.